src/util/path.js

```diff
@@ -729,6 +741,70 @@
     return result;
   };
 
+  /**
+   *
+   * Converts points to a smooth SVG path
+   * @param {{ x: number,y: number }[]} points Array of points
+   * @param {number} [correction] Apply a correction to the path (usually we use `width / 1000`). If value is undefined 0 is used as the correction value.
+   * @return {(string|number)[][]} An array of SVG path commands
+   */
+  function getSmoothPathFromPoints(points, correction) {
+    var path = [], i,
+        p1 = new fabric.Point(points[0].x, points[0].y),
+        p2 = new fabric.Point(points[1].x, points[1].y),
+        len = points.length, multSignX = 1, multSignY = 0, manyPoints = len > 2;
+    correction = correction || 0;
+
+    if (manyPoints) {
+      multSignX = points[2].x < p2.x ? -1 : points[2].x === p2.x ? 0 : 1;
+      multSignY = points[2].y < p2.y ? -1 : points[2].y === p2.y ? 0 : 1;
+    }
+    path.push(['M', p1.x - multSignX * correction, p1.y - multSignY * correction]);
+    for (i = 1; i < len; i++) {
+      if (!p1.eq(p2)) {
+        var midPoint = p1.midPointFrom(p2);
+        // p1 is our bezier control point
+        // midpoint is our endpoint
+        // start point is p(i-1) value.
+        path.push(['Q', p1.x, p1.y, midPoint.x, midPoint.y]);
+      }
+      p1 = points[i];
+      if ((i + 1) < points.length) {
+        p2 = points[i + 1];
+      }
+    }
+    if (manyPoints) {
+      multSignX = p1.x > points[i - 2].x ? 1 : p1.x === points[i - 2].x ? 0 : -1;
+      multSignY = p1.y > points[i - 2].y ? 1 : p1.y === points[i - 2].y ? 0 : -1;
+    }
+    path.push(['L', p1.x + multSignX * correction, p1.y + multSignY * correction]);
+    return path;
+  }
+
+
+  /**
+   * Converts points to a smooth SVG path.
+   * This utility calculates the path's correction value from `points`.
+   * @param {{ x: number,y: number }[]} points 
+   * @return {(string|number)[][]} An array of SVG path commands
+   */
+  function getSmoothPathFromPointsWithCorrection(points) {
+    //  we infer `correction` by calculating the path's width
+    var start = points[0].x;
+    var minMax = points.reduce(
+      function (prev, curr) {
+        return {
+          min: Math.min(curr.x, prev.min),
+          max: Math.max(curr.x, prev.max)
+        };
+      },
+      { min: start, max: start }
+    );
+    var width = minMax.max - minMax.min;
+    var correction = width / 1000;
+    return getSmoothPathFromPoints(points, correction);
+  }
```

Comment:
we don't need this automatic correction, at all.
this width can't be inferred from the point, is just the brush chosen width, the brush stroke, and has nothing to do with the final size of the path.

The correction is applied for when a final user does a single click and to avoid get a dimensionless path we move a tiny tiny bit the first  and last point.

This is specific of the brush and does not need to be automatic.

Reply:
REVERTED adba2157